pages/k8s/openstack-integration.md

```diff
@@ -256,13 +280,24 @@ Keystone. This is covered in detail in the [Keystone and LDAP documentation][lda
 
 ### Upgrading the integrator charm
 
-The openstack-integrator is not specifically tied to the version of Charmed Kubernetes installed and may
+The openstack-integrator has not specifically been tied to the version of Charmed Kubernetes installed and may
 generally be upgraded at any time with the following command:
 
 ```bash
 juju refresh openstack-integrator
 ```
 
+The 1.29/stable release of `openstack-integrator` replaces the relation for using Octavia as a loadbalancer for the API Service. 
+The 1.29/stable release of `kubernetes-control-plane` drops the responsibility of deploying `cinder-csi`` and the `openstack-controller-manager`
+In order to upgrade the control-plane and worker charms, follow this process:
+
+* 1) Upgrade the openstack-integrator charm with `juju refresh openstack-integrator --switch --channel=1.29/stable`
+* 2) Relate to the control-plane application with `juju relate openstack-integrator:lb-consumer kubernetes-control-plane:loadbalancer-external`
+* 3) Deploy and migrate to the `openstack-cloud-controller` charm (See its [charm docs][openstack-cloud-controller-readme] for details)
+* 4) Deploy and migrate to the `cinder-csi` charm (See its [charm docs][cinder-csi-readme] for details)
+* 5) Remove the loadbalancer relation to the control-plane with `juju remove-relation openstack-integrator:loadbalancer kubernetes-control-plane:loadbalancer`
```

Comment:
This looks quite odd in Markdown. Perhaps using an ordered list would work better?
```suggestion
1. Upgrade the openstack-integrator charm with `juju refresh openstack-integrator --switch --channel=1.29/stable`
2. Relate to the control-plane application with `juju relate openstack-integrator:lb-consumer kubernetes-control-plane:loadbalancer-external`
3. Deploy and migrate to the `openstack-cloud-controller` charm (See its [charm docs][openstack-cloud-controller-readme] for details)
4. Deploy and migrate to the `cinder-csi` charm (See its [charm docs][cinder-csi-readme] for details)
5. Remove the loadbalancer relation to the control-plane with `juju remove-relation openstack-integrator:loadbalancer kubernetes-control-plane:loadbalancer`
```